salt/edx/run_ansible.sls

```diff
@@ -94,3 +94,13 @@ run_ansible:
     - unless: {{ salt.pillar.get('edx:skip_ansible', False) }}
     - env:
         HOME: /root
+
+{% if 'edx-worker' in salt.grains.get('roles') %}
+restart_edx_worker_service:
+  supervisord.running:
+    - name: all
+    - restart: True
+    - bin_env: '/edx/bin/supervisorctl'
+    - onchanges_in:
```

Comment:
This would be `onchanges` for this case. The `_in` suffix would be if you added it to the conf file state id. Also, the argument should use the state id for that step.

```
    - onchanges:
      - file: place_ansible_environment_configuration
```

Reply:
This should also have a `require` statement for the `run_ansible` step.